arangod/Agency/CleanOutServer.cpp

```diff
@@ -345,6 +345,7 @@ bool CleanOutServer::start(bool& aborts) {
       addPreconditionServerHealth(*pending, _server, "GOOD");
       addPreconditionUnchanged(*pending, failedServersPrefix, failedServers);
       addPreconditionUnchanged(*pending, cleanedPrefix, cleanedServers);
+      addPreconditionUnchanged(*pending, planVersion, _snapshot(planVersion).slice());
```

Comment:
isn't there any other way than to check on planVersion to be unchanged?

Plan version is used for everything that happens in the cluster, if it has anything to do with this server or not.
So this does not scale well...

Reply:
That is correct, however, it is only `CleanOutServer`, so that it is possibly just delaying the jobs from starting. It should not impact the user experience in significant ways, as the admins do not expect the performance of this process to be of importance.

Reply:
No there is not, because we have no index for `dbserver -> collection`.